clients.go

```diff
@@ -83,6 +96,10 @@ func NewClients(config ClientsConfig) (*Clients, error) {
 	return c, nil
 }
 
+func (c *Clients) DynClient() dynamic.Interface {
```

Comment:
I think going with the interface is the better approach in cases where the interface is actually part of another interface. I noticed problems in our e2e libs where some interface definition required some pointer and compiling failed because the required interface was interface and not pointer. 🙈

Reply:
Agreed. I'll change that in follow up PR.

Reply:
https://github.com/giantswarm/k8sclient/pull/6